exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScan.java

```diff
@@ -115,6 +115,8 @@
   private List<RowGroupInfo> rowGroupInfos;
   private Metadata.ParquetTableMetadataBase parquetTableMetadata = null;
   private String cacheFileRoot = null;
+  private int batchSize;
+  private static final int DEFAULT_BATCH_LENGTH = 256 * 1024;
```

Comment:
Now I guess I understand what you mean.  You want to cap store.parquet.record_batch_size to 256K. And set DEFAULT_BATCH_LENGTH =256k in ParquetGroupScan. At execution time, you pick min of batch_size and value from option, which will be no greater than the option value. 

If that's correct, can we remove DEFAULT_BATCH_LENGTH in ParquetGroupScan. In stead, use the batch_size specified in the new option you added for NON-LIMIT case?